electric-sql/electric
# fix(client) Various fixes to run generator and starter scripts on windows

Included fixes:
- Prefix zod type generator provider with node for exe on windows so that its run with node
- Conditionally escape quotes in docker command on windows, this has to be conditional as it breaks Posix
- Execute the pg-migrations command with npx not yarn

The fixes to the starter are also applied to the web-wa-sqlite example.

---

### To test:

```sh
git clone git@github.com:electric-sql/electric.git && cd electric
git checkout samwillis/windows-fix
pnpm install
cd clients/typescript
pnpm build && pnpm pack
cd ../../examples/web-wa-sqlite/
npm uninstall electric-sql
npm uninstall @prisma/client @electric-sql/prisma-generator prisma  # These are no longer required but we haven't updated the example yet 
npm install ../../clients/typescript/electric-sql-0.8.1.tgz
npm backend:up
npm db:migrate
npm client:generate
npm start
```

If for some reason the client won't build on windows, this is a ready made build from this branch: 
[electric-sql-0.8.1.tgz](https://github.com/electric-sql/electric/files/13635086/electric-sql-0.8.1.tgz) - just try it with the web-wa-sqlite *in this branch*
